client/components/credit-card-form-fields/index.jsx

```diff
@@ -59,54 +80,158 @@ class CreditCardFormFields extends React.Component {
 			[ fieldName ]: maskField( fieldName, previousValue, nextValue ),
 		};
 
-		this.props.onFieldChange( rawDetails, maskedDetails );
+		onFieldChange( rawDetails, maskedDetails );
+	}
+
+	handlePhoneFieldChange = ( { value, countryCode } ) => {
+		this.setState(
+			{
+				phoneCountryCode: countryCode,
+			},
+			() => {
+				this.updateFieldValues( 'phone-number', value );
+			}
+		);
+	};
+
+	handleFieldChange = event => {
+		const { name: fieldName, value: nextValue, options, selectedIndex } = event.target;
+
+		const newState = {};
+
+		if ( fieldName === 'country' ) {
+			newState.countryCode = nextValue;
+			newState.countryName = options[ selectedIndex ].text;
+			if ( ! this.state.phoneCountryCode ) {
+				newState.phoneCountryCode = nextValue;
+			}
+		}
+
+		this.setState( newState, () => this.updateFieldValues( fieldName, nextValue ) );
 	};
 
+	shouldRenderEbanx() {
+		const { countryCode } = this.state;
+		return isEbanx( countryCode );
+	}
+
+	renderEbanxFields() {
+		const { translate, countriesList } = this.props;
+		const { countryCode, phoneCountryCode, countryName } = this.state;
+
+		return [
+			<span key="ebanx-required-fields" className="credit-card-form-fields__info-text">
+				{ translate( 'The following fields are also required for payments in %(countryName)s', {
+					args: {
+						countryName,
+					},
+					context: 'Info text for extra ebanx fields in credit card form',
+				} ) }
+			</span>,
+
+			this.createField( 'document', Input, {
+				label: translate( 'Taxpayer Identification Number', {
+					context:
```

Comment:
Let's switch to a translator comment here, and lose the EBANX mention which is irrelevant for translators.